BWEnv/src/controller.cc

```diff
@@ -434,17 +434,71 @@ int8_t Controller::handleCommand(int command, const std::vector<int>& args,
       draw_cmds_.push_back({cmd, str});
       return CommandStatus::SUCCESS;
     }
-    case Commands::COMMAND_USER:
+    case Commands::COMMAND_USER: {
       auto type = args[0];
       auto second = args.begin() + 1;
       auto last = args.end();
       std::vector<int> user_args(second, last);
 
       return handleUserCommand(type, user_args);
     }
+    case Commands::COMMAND_OPENBW: {
+      // includes type
+      static std::unordered_map<int, int> obw_argcount = {
+        {OBWCommands::KILL_UNIT, 3}, {OBWCommands::SPAWN_UNIT, 5},
+      };
+      if (!check_args(obw_argcount[command])) return status;
+      auto type = args[0];
+      auto second = args.begin() + 1;
+      auto last = args.end();
+      std::vector<int> user_args(second, last);
+
+      return handleOpenBWCommand(type, user_args);
+    }
+    }
+  }
+  Utils::bwlog(output_log, "Invalid command: %d", command);
+  return CommandStatus::UNKNOWN_COMMAND;
+}
+
+int8_t Controller::handleOpenBWCommand(int command, const std::vector<int>& args)
+{
+  int8_t status = CommandStatus::OPENBW_NOT_IN_USE;
+  #ifndef OPENBW_BWAPI
+  return status;
+  #else
+  switch (command)
+  {
+  case OBWCommands::KILL_UNIT: {
+    auto u = BWAPI::Broodwar->getUnit(args[0]);
+    if (u == nullptr)
+    {
+      status = CommandStatus::INVALID_UNIT;
+      return false;
```

Comment:
Do you mean `return status;` here? There's a couple more to fix later down the road.